pyoomph/c_output.py

```diff
@@ -339,6 +345,20 @@ def emit_funcdef(
             self->len += src->len;
         }
 
+        void meth_%(type_cname)s_insert(struct class_%(type_cname)s *self, int64_t index, %(itemtype)s val)
+        {
+            class_%(type_cname)s_ensure_alloc(self, self->len + 1);
```

Comment:
Maybe move this line just before `memmove` to make it not look any more buggy than it is?